docs/proposals-accepted/202304-cardinalitymanagment.md

```diff
@@ -0,0 +1,190 @@
+- https://mentorship.lfx.linuxfoundation.org/project/dbce5279-d029-46f3-b117-9e9dd7f84bd6
+
+### Why
+
+At present, multiple product teams send their metrics to Thanos. However, when product teams send metrics with high cardinality to Thanos, it can overwhelm the Store Gateway and Queriers, preventing any teams from running queries. To prevent this issue from occurring, we should have a cardinality management support in Thanos. This would allow teams to monitor the cardinality of both labels and metrics.
+
+### Pitfalls of current solutions
+
+Currently, Thanos receiver supports the TSDB stats API on remote write addresses, which provides information about series cardinality. However, this API can only return a limited number of results per query, typically no more than 10. This limitation makes it difficult to track arbitrary metrics and may lead to incomplete or inaccurate cardinality data. To address this issue, there should be some Top-K selector that can return the stats of the top K entries, allowing teams to more effectively monitor cardinality and identify potential issues.
+
+### Goals
+
+Implement APIs exposing series cardinalities in Thanos.
+
+### Proposal
+
+We will be Developing a new API in Thanos to calculate series cardinality through Thanos Receiver.
+
+#### API Design
+
+The new API will have the following endpoint-
+
+`GET /api/v1/cardinality`
+
+It will retrieve the series cardinality information.
+
+The API will accept the following query parameters:
+
+- topK : (optional) An integer specifying the top K results to return for each category. Default is 10.
+- selector : (optional) A PromQL selector that will be used to filter series that must be analyzed.
+- focusLabel : (optional) A string representing the label name for which you want to retrieve the series count by its unique values. When provided, the API response will include the `seriesCountByFocusLabelValue` field, containing the unique values for the specified label and their corresponding series counts.
+
+The API response will contain
+
+- `totalSeries`: Total number of series.
+- `totalLabelValuePairs`: Total number of label-value pairs.
+- `seriesCountByMetricName`: List of objects containing metric name and its corresponding series count.
+- `seriesCountByLabelName`: List of objects containing label name and its corresponding series count.
+- `seriesCountByFocusLabelValue`: List of objects containing focus label-value and its corresponding series count.
+- `seriesCountByLabelValuePair`: List of objects containing label-value pair and its corresponding series count.
+- `labelValueCountByLabelName`: List of objects containing label name and its corresponding label value count.
+
+For example, if we have the following dataset:
+
+```
+{__name__="metricA", instance="A", cluster="us"}
+{__name__="metricA", instance="B", cluster="eu"}
+{__name__="metricB", instance="C", cluster="us"}
+```
+
+Then, API response would be
```

Comment:
Can you please provide a full example? You give the response but you didn't mention the request parameters, right? Then it is still kind of confusing. Several examples with different input would great.